blockio-uring.cabal

```diff
@@ -45,7 +45,7 @@ library
     , primitive  ^>=0.9
     , vector     ^>=0.13
 
-  pkgconfig-depends: liburing >=2.0 && <2.7
+  pkgconfig-depends: liburing >=2.0 && <3
```

Comment:
The tight upper bound on an unreleased `2.x` version is intentional: it means that we have to explicit about which versions we support, so that the build doesn't break under our feet when a new version is released. In general, I think tight bounds are better when defining dependencies, so I'd prefer `<2.9`

Reply:
(I'm biased towards tight bounds because it's unlikely that we can truly guarantee that libraries will work with newer version in the future. What we can do is guarantee that it works with currently released versions)

Reply:
I thought we had said during the call that we can be more liberal with the upper bound here, but I might have misunderstood. I'm fine with `<2.9`. A little bit of extra work to increase bounds, but also ensures that we run CI again on each version bump.